packages/server/src/api/routes/public/tests/Request.ts

```diff
@@ -0,0 +1,110 @@
+import { User, Table, SearchFilters, Row } from "@budibase/types"
+import { HttpMethod, MakeRequestResponse, generateMakeRequest } from "./utils"
+import TestConfiguration from "../../../../tests/utilities/TestConfiguration"
+import { Expectations } from "../../../../tests/utilities/api/base"
+
+type RequestOpts = { internal?: boolean; appId?: string }
+
+type PublicAPIExpectations = Omit<Expectations, "headers" | "headersNotPresent">
+
+export class PublicAPIRequest {
+  private makeRequest: MakeRequestResponse | undefined
+  private appId: string | undefined
+  private _tables: PublicTableAPI | undefined
+  private _rows: PublicRowAPI | undefined
+  private _apiKey: string | undefined
```

Comment:
This is a bit of a mess, and makes a lot of the rest of the code of this class quite awkward. I know it somewhat mirrors `TestConfiguration` but I eventually want us to fix that, too.

I'd recommend doing something like this:

```typescript
// The name `PublicAPIRequest` feels incorrect, it's not a request, it's
// a bridge to other APIs such as `tables` and `rows`.
export class PublicAPI {
  private makeRequest: MakeRequestResponse
  private apiKey: string
  private appId?: string

  tables: PublicTableAPI
  rows: PublicRowAPI

  constructor(config: TestConfiguration, user: User, opts?: RequestOpts) {
    this.apiKey = await config.generateApiKey(user._id)
    this.makeRequest = generateMakeRequest(this.apiKey, opts)
    this.appId = opts?.appId
    this.tables = new PublicTableAPI(this)
    this.rows = new PublicRowAPI(this)
  }
}
```

Then I think it would be ideal if this lived alongside `config.api` as `config.publicApi`. That would make it much nicer when used, rather than having to create these object by hand when needed.